Detections/SecurityEvent/TimeSeriesAnomaly-ProcessExecutions.txt

```diff
@@ -0,0 +1,30 @@
+// Name: Time Series Anomaly detection for Process execution frequency
+// Description: This query will identify anamalous spike in frequency of executions of sensitive processes which are often leveraged as attack vector. 
+// The query leverages KQL built-in anomaly detection algorithm find large deviations from baseline pattern. 
+// Any sudden increase in execution frequency of sensitice procecesss should be further investigated for presence of malicious activity by threat actor.
+//
+// Id: 2c55fe7a-b06f-4029-a5b9-c54a2320d7b8
+//
+// DataSource: #SecurityEvent
+//
+// Tactics: #Execution, #InitialAccess
+//
+let starttime = 2d;
+let endtime = 1d;
+let timeframe = 1h;
+let TotalEventsThreshold = 5;
+let ExeList = dynamic(
+["powershell.exe","cmd.exe","psexec.exe","cacls.exe","rundll.exe"]);
+let TimeSeriesData= 
+SecurityEvent
+| where EventID == "4688" | extend Process = tolower(Process)
+| where TimeGenerated between (ago(starttime)..ago(endtime))
+| where Process in (ExeList)
+| project TimeGenerated,Computer, AccountType, Account, Process
+| make-series Total=count() on TimeGenerated from ago(starttime) to ago(endtime) step timeframe by Process;
+TimeSeriesData
+| extend (anomalies, score, baseline) = series_decompose_anomalies(Total, 1.5, -1, 'linefit')
+| mv-expand Total to typeof(double), TimeGenerated to typeof(datetime), anomalies to typeof(double),score to typeof(double), baseline to typeof(long)
+| where anomalies > 0
+| project Process,TimeGenerated, Total, baseline, anomalies, score
+| where Total > TotalEventsThreshold
```

Comment:
Let's join this back with the original data to get the CommandLine, full path and account from the event.  Also, verify the timestamp that is produced, it looked off to me. :)

Reply:
Added relevant fields in output view along with fix for timestamphour in the new commit

Reply:
Looks good